### PR TITLE
Migrate to Swift 5

### LINKED ISSUE #79: Modify WordPressKit for Swift 5 support
### Expected behavior

We should adapt the framework to support Swift 5.

### Actual behavior

At the moment the framework uses Swift 4.2.

### Steps to reproduce the behavior

For all targets in the project, `SWIFT_VERSION` is currently set to 4.2.

##### WordPressKit Environment
 - WordPressKit Version : `WordPressKit` v1.8.0 & before
 - OS : iOS 12.x
 - Device : all devices
 - Xcode Version : Xcode 10.2 beta

##### Additional Context
[Swift 5 Release Notes for Xcode 10.2 beta](https://developer.apple.com/documentation/xcode_release_notes/xcode_10_2_beta_release_notes/swift_5_release_notes_for_xcode_10_2_beta)

### LINKED ISSUE #149: Bug: Logging of Social Login exposes OAuth Bearer token to log files
### Expected behavior

Log entries shouldn't include sensitive data like bearer tokens.

### Actual behavior

When logging in with social login the entire response object is dumped to the log and exposes the bearer token to the app's log file.

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/c834839a899732dc3c7092f01626fa93b41b475f/WordPressKit/WordPressComOAuthClient.swift#L274

```
2019/04/27 12:57:11:934  Received Social Login Oauth response: {
    data =     {
        "bearer_token" = "manually redacted by me for demo purposes";
        "token_links" =         (
            "https://jetpack.com/remote-login.php?wpcom_remote_login=validate&wpcomid=xx&token=xx&host=https%3A%2F%2Fjetpack.com"
        );
    };
    success = 1;
}
```

### Steps to reproduce the behavior

1. Use Social Login and look at the log file.

### PR DESCRIPTION
**Pull request by mindgraffiti**
### Description

Fixes #79
Fixes #149 

Test this PR by checking out https://github.com/wordpress-mobile/WordPress-iOS/pull/13641

In this PR:

- updates shared pod schemes to 11.3
- `pod update` to get the latest acceptable changes for the internal (WordPress) pods
- update the dependent framework OHHTTPStubs from 6.1.0 -> 9.0
- remove `OH` class prefix from code that uses OHHTTPStubs
- update unit test target to use Swift 5
- update main project to use Swift 5
- enable base internationalization
- update podspec to require Swift version 5
- add a bug fix to logged info

### Known issues
A warning will still remain in this project. Alamofire needs migrated from 4.x to 5.x in order to fully adopt Swift 5 in the project. A github issue will be filed for this.

### Testing Details

- `bundle exec pod install` 
- build and run the project

- [ ] Please check here if your pull request includes additional test coverage.

### CODE REVIEW

File: WordPressKit/PluginDirectoryEntry.swift [hunk 1]
```diff
@@ -83,16 +83,16 @@ extension PluginDirectoryEntry: Codable {
         lastUpdated = try? container.decode(Date.self, forKey: .lastUpdated)
         rating = try container.decode(Int.self, forKey: .rating)
 
-        let icons = try? container.decodeIfPresent([String: String].self, forKey: .icons)
+        let icons = ((try? container.decodeIfPresent([String: String].self, forKey: .icons)) as [String : String]??)
```
Comment (Gio2018) [inline WordPressKit/PluginDirectoryEntry.swift]: is there a specific reason for casting to `[String: String]?` ? `decodeIfPresent` should return a `[String: String]` value or `nil`, depending if that value exists (and is not nil) or not, so if we leave this statement unchanged, can the following just be 
 `icon = icons?["2x"].flatMap(URL.init(string:))`?

or we could also:

``if let icons = try container.decodeIfPresent([String: String].self, forKey: .icons) {
            icon = icons["2x"].flatMap(URL.init(string:))
 }``

Comment (diegoreymendez) [inline WordPressKit/PluginDirectoryEntry.swift]: I think these two lines should be:

```swift
let icons = try? container.decodeIfPresent([String: String].self, forKey: .icons)
icon = icons?["2x"].flatMap(URL.init(string:))
```

File: WordPressKit/PluginDirectoryEntry.swift [hunk 2]
```diff
@@ -83,16 +83,16 @@ extension PluginDirectoryEntry: Codable {
...
         icon = icons??["2x"].flatMap(URL.init(string:))
 
         // If there's no hi-res version of the banner, the API returns `high: false`, instead of something more logical,
         // like an empty string or `null`, hence the dance below.
         let banners = try? container.nestedContainer(keyedBy: BannersKeys.self, forKey: .banners)
 
-        if let highRes = try? banners?.decodeIfPresent(String.self, forKey: .high) {
+        if let highRes = ((try? banners?.decodeIfPresent(String.self, forKey: .high)) as String??) {
```
Comment (diegoreymendez) [inline WordPressKit/PluginDirectoryEntry.swift]: I think this whole block should be:

```swift
if let highRes = try? banners?.decodeIfPresent(String.self, forKey: .high) {
    banner = URL(string: highRes)
} else if let lowRes = try? banners?.decodeIfPresent(String.self, forKey: .low) {
    banner = URL(string: lowRes)
} else {
    banner = nil
}
```

File: WordPressKit/WordPressComOAuthClient.swift [hunk 3]
```diff
@@ -272,7 +272,7 @@ public final class WordPressComOAuthClient: NSObject {
             .responseJSON(completionHandler: { response in
                 switch response.result {
                 case .success(let responseObject):
-                    DDLogVerbose("Received Social Login Oauth response: \(self.cleanedUpResponseForLogging(responseObject as AnyObject? ?? "nil" as AnyObject))")
```
Comment (diegoreymendez) [inline WordPressKit/WordPressComOAuthClient.swift]: Is this removal related to the migration or something else?

I tried using the original line and it seems to work fine in Swift 5.

File: WordPressKit/WordPressOrgXMLRPCValidator.swift [hunk 4]
```diff
@@ -325,7 +325,7 @@ open class WordPressOrgXMLRPCValidator: NSObject {
             guard let data = data,
                 let responseString = String(data: data, encoding: String.Encoding.utf8),
                 let parser = WordPressRSDParser(xmlString: responseString),
-                let endpoint = (try? parser.parsedEndpoint()),
+                let endpoint = (((try? parser.parsedEndpoint()) as String??)),
```
Comment (diegoreymendez) [inline WordPressKit/WordPressOrgXMLRPCValidator.swift]: I think this line should be:

```swift
let xmlrpc = try? parser.parsedEndpoint(),
```

And the immediate next line should be deleted.  This one:

```swift
let xmlrpc = endpoint,
```

File: WordPressKit/PluginDirectoryEntry.swift (hunk 2 above)
Comment (Gio2018) [inline WordPressKit/PluginDirectoryEntry.swift]: Maybe we could do:
`if let highRes = try banners?.decodeIfPresent(String.self, forKey: .high) {
            banner = URL(string: highRes)
        }`

File: WordPressKit/WordPressOrgXMLRPCValidator.swift [hunk 5]
```diff
@@ -325,7 +325,7 @@ open class WordPressOrgXMLRPCValidator: NSObject {
...
                 let xmlrpc = endpoint,
                 let xmlrpcURL = URL(string: xmlrpc)
                 else {
```
Comment (Gio2018) [inline WordPressKit/WordPressOrgXMLRPCValidator.swift]: maybe the whole guard statement can be simplified like so?

`guard let data = data,`
                `let responseString = String(data: data, encoding: String.Encoding.utf8),`
                `let parser = WordPressRSDParser(xmlString: responseString),`
                `let xmlrpc = try? parser.parsedEndpoint(),`
                `let xmlrpcURL = URL(string: xmlrpc)`
                `else {`
                    `failure(WordPressOrgXMLRPCValidatorError.invalid as NSError)`
                   ` return`
            `}`

File: WordPressKit/WordPressComOAuthClient.swift (hunk 3 above)
Comment (mindgraffiti) [reply WordPressKit/WordPressComOAuthClient.swift]: Good question - this removal is the bugfix for #149. Social logins should not expose the bearer token.

Comment (mindgraffiti) [reply WordPressKit/WordPressComOAuthClient.swift]: I made this addition because it seemed important to fix.

File: WordPressKit/PluginDirectoryEntry.swift (hunk 1 above)
Comment (mindgraffiti) [reply WordPressKit/PluginDirectoryEntry.swift]: No there is not. I let Xcode do the migrating and overlooked how odd that change really was. Correcting it now :)

**Comment by mindgraffiti** [comment]
@Gio2018 @diegoreymendez ready for a second review! I see what the `icon` property was attempting to. Please check the re-written code. This pod can be tested against this WPiOS branch: https://github.com/wordpress-mobile/WordPress-iOS/pull/13641. You may have to `bundle exec pod update WordPressKit` to force the pod to update with the latest code in this branch.

To test the changed `icon` code, open `PluginDirectoryEntry.swift` in WPiOS and add a breakpoint to L87-L88. Then navigate to the "Plugins" page from your preferred site.
